tests/integration/031_vault_credentials/conf.tf

```diff
@@ -0,0 +1,15 @@
+terraform {
+  backend "local" {
+  }
+  required_providers {
+    env0 = {
+      source = "terraform-registry.env0.com/env0/env0"
+    }
+  }
+}
+
+provider "env0" {}
+
+variable "second_run" {
```

Comment:
why do we need this variable?

Reply:
I actually want to use it to test updates.
But hadn't applied it to the test. Adding it now.

Reply:
See this is an example:
https://github.com/env0/terraform-provider-env0/pull/783/commits/eef71ed878fdafeb20fd7f6e1fbc945233007470

Reply:
it looks like we have some failing tests

Reply:
once you fix those you are good to go

Reply:
yeah I just found out update was never tested.
I'm adding the tests, and fixing the cause for it failing.
I'll push once the tests path. Thanks!

Reply:
never tested not just for OIDC... but all AWS, GCP and Azure for all credential types...

Reply:
Can you elaborate on how was it missed? So I will be able to catch it faster in the future

Reply:
These are old tests, written before I started working on this project.
I noticed that` var.second_run` wasn't used in any of the tests. If it's not used, "update" isn't tested.
For "update" to work properly I had to add "omitempty" to some of the structures (this is why vault failed to pass).

In other words, there were bugs in the provider that were not detected because tests were missing.

I added the tests and fixed the bugs. Now it should all work as expected.

Reply:
Alright thanks 😊